src/appleseed/renderer/kernel/lighting/lighttree.cpp

```diff
@@ -226,26 +259,9 @@ void LightTree::sample(
     }
 
     const size_t item_index = m_nodes[node_index].get_item_index();
-    const size_t source_index = m_items[item_index].m_light_source_index;
-    light_type = m_light_sources[source_index]->get_type();
 
-    switch (light_type)
-    {
-      // m_light_source_index corresponds to the m_light_tree_lights
-      // index of the BackwardLightSampler in case of NPL, whereas the EMT index
-      // needs to be stored separately for now as it is not in the same
-      // vector with other light tree compatible lights.
-      case LightSource::NonPhysicalLightType:
-        light_index = m_items[item_index].m_light_source_index;
-        break;
-
-      case LightSource::EmittingTriangleType:
-        light_index = m_items[item_index].m_external_source_index;
-        break;
-
-      default:
-        assert(!"Unexpected light type.");
-    }
+    light_type = m_items[item_index].m_light_type;
```

Comment:
Factorize `m_items[item_index]`. Even though it's likely that the compiler do optimizes this, we're much better off relying as little as possible on the compiler's decisions.